### PR TITLE
Storybook: Add controls to button story

### LINKED ISSUE #54161: Add controls to `Button` story
Add support for [controls](https://storybook.js.org/docs/react/essentials/controls) for the various props in a story.

The current existing `Variants` story should be renamed to `Examples` and a new story should be created that shows only a single instance of the `Button` and has controls attached to it.

### PR DESCRIPTION
**Pull request by Pomierski**
**What this PR does / why we need it**: This PR adds new story in storybook for `Button` component with configured [controls](https://storybook.js.org/docs/react/essentials/controls). Current "variants" story was renamed to "examples" as requested in issue description.

**Which issue(s) this PR fixes**: https://github.com/grafana/grafana/issues/54161

Fixes #54161

**Special notes for your reviewer**:

### CODE REVIEW

**Review by jackw — APPROVED** [review]
Thanks so much for contributing to Grafana @Pomierski ! 🥇 

Code looks good, and stories behave as expected. I've left a couple of comments but let's wait on @JoaoSilvaGrafana for final review. 🙂

File: packages/grafana-ui/src/components/Button/Button.story.tsx
Comment (jackw) [inline packages/grafana-ui/src/components/Button/Button.story.tsx:23]: nit: Not a biggie but we could pull `sizes` out of `Examples` story and reference it here.

File: packages/grafana-ui/src/components/Button/Button.story.tsx
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/Button/Button.story.tsx:111]: We should add the other defaults here for the controls

```suggestion
Default.args = {
  children: 'Example button',
  size: 'md',
  variant: 'primary',
  fill: 'solid',
};
```

File: packages/grafana-ui/src/components/Button/Button.story.tsx
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/Button/Button.story.tsx:35]: Why did we disable these on the controls? I think we could let them be set to see how they work

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Thank you so much for the submission! 👏 

Added a few comments about things I think we could add or remove, let me know what you think 😄

File: packages/grafana-ui/src/components/Button/Button.story.tsx
Comment (Pomierski) [reply packages/grafana-ui/src/components/Button/Button.story.tsx]: Thanks for review 😄 Applied suggestion with other changes. I used ComponentStory interface and removed unpacking of args.

Comment (Pomierski) [reply packages/grafana-ui/src/components/Button/Button.story.tsx]: Renamed Default to Basic

File: packages/grafana-ui/src/components/Button/Button.story.tsx
Comment (Pomierski) [reply packages/grafana-ui/src/components/Button/Button.story.tsx]: Done

File: packages/grafana-ui/src/components/Button/Button.story.tsx
Comment (Pomierski) [reply packages/grafana-ui/src/components/Button/Button.story.tsx]: Thanks for review 😄  I removed expanded controls and excluded className

File: packages/grafana-ui/src/components/Button/Button.story.tsx
Comment (Pomierski) [reply packages/grafana-ui/src/components/Button/Button.story.tsx]: Done

File: packages/grafana-ui/src/components/Button/Button.story.tsx
Comment (Pomierski) [reply packages/grafana-ui/src/components/Button/Button.story.tsx]: `tooltip` and `tooltipPlacement` props seems to be only used in `LinkButton` component

Comment (JoaoSilvaGrafana) [reply packages/grafana-ui/src/components/Button/Button.story.tsx]: Ah yes you are totally right!

**Review by JoaoSilvaGrafana — APPROVED** [review]
Great job on this 🎉

**Pull request merged**
